### ISSUE TITLE
CTKLabel configure with Chinese string issue

### ISSUE THREAD

**Issue by Corporation001** [TomSchimansky/CustomTkinter#126]
Hello, If I configure CTKLabel with Chinese string, it will not show all the text.
Like the follow picture:
![image](https://user-images.githubusercontent.com/106279144/171036496-626c61f9-f24a-4560-9404-e1993e06015e.png)
Correct Label will behave like:
![image](https://user-images.githubusercontent.com/106279144/171036595-5bb5b96c-188b-4621-b26d-7b0525ae0cb8.png)
hope you can fix it!
Sincerely.

**Comment by felipetesc** [comment]
Hello. Can you post the string itself and not an image, please. TY!

**Comment by felipetesc** [comment]
I will create some tests. By the way, did you added a blank space before the word ? I'm only curious about the result.I will create some tests. By the way, did you added a blank space before the word ? I'm only curious about the result.

**Comment by Corporation001** [comment]
The string is "沒有選擇檔案".
Thanks!
about add a blank,I will have a try.

**Comment by Corporation001** [comment]
> I will create some tests. By the way, did you added a blank space before the word ? I'm only curious about the result.

Whether I add a blank or not, I will get the same result.

**Comment by felipetesc** [comment]
I've tested this, and it works (Can you try please) :

```python

import customtkinter
import tkinter



class TestApp(customtkinter.CTk):
    def __init__(self, str_to_test):
        super().__init__()
        self.num_of_frames = 0
        # self.state('withdraw')
        self.title("Webcam sample")
        self.geometry("800x600")
        self.create_ui(str_to_test)

    def create_ui(self, str_to_test):
        self.main_frame = customtkinter.CTkFrame(self)
        self.main_frame.pack(fill="both", expand=True)
        self.label_with_chinese = customtkinter.CTkLabel(self.main_frame, text=str_to_test)
        self.label_with_chinese.pack(fill="both", expand=True)


str_with_issue = "沒有選擇檔案"
app1 = TestApp(str_with_issue)
app1.mainloop()


str_without_issue = "你好我是 个有趣的标签"
app2 = TestApp(str_without_issue)
app2.mainloop()

```
Right now I'm using Garuda Linux 64. 
Also, I found this post:
https://stackoverflow.com/questions/60136361/chinese-text-in-tkinter-python 
Perhaps, you need to add a custom font for the label, and call it. What's the default font from your system. Did You try using only Tkinter ? Because, if You find the same issue with tkinter, perhaps it's not something from ctk.

**Comment by Corporation001** [comment]
> I've tested this, and it works (Can you try please) :
> 
> ```python
> import customtkinter
> import tkinter
> 
> 
> 
> class TestApp(customtkinter.CTk):
>     def __init__(self, str_to_test):
>         super().__init__()
>         self.num_of_frames = 0
>         # self.state('withdraw')
>         self.title("Webcam sample")
>         self.geometry("800x600")
>         self.create_ui(str_to_test)
> 
>     def create_ui(self, str_to_test):
>         self.main_frame = customtkinter.CTkFrame(self)
>         self.main_frame.pack(fill="both", expand=True)
>         self.label_with_chinese = customtkinter.CTkLabel(self.main_frame, text=str_to_test)
>         self.label_with_chinese.pack(fill="both", expand=True)
> 
> 
> str_with_issue = "沒有選擇檔案"
> app1 = TestApp(str_with_issue)
> app1.mainloop()
> 
> 
> str_without_issue = "你好我是 個有趣的標簽"
> app2 = TestApp(str_without_issue)
> app2.mainloop()
> ```
> 
> Right now I'm using Garuda Linux 64. Also, I found this post: https://stackoverflow.com/questions/60136361/chinese-text-in-tkinter-python Perhaps, you need to add a custom font for the label, and call it. What's the default font from your system? Did You try using only Tkinter ? Because, if You find the same issue with tkinter, perhaps it's not something from ctk.

In regular tkinter, it works correctly.
I useing Windows 11,font is default by system.
I modded your code, it have a button to confugure the label.
It will cause the problem I have.
`# -*- coding: utf-8 -*-
"""
Created on Tue May 31 02:37:31 2022

@author: ecoco
"""

import customtkinter
import tkinter



class TestApp(customtkinter.CTk):
    def __init__(self, str_to_test):
        super().__init__()
        self.num_of_frames = 0
        # self.state('withdraw')
        self.title("Webcam sample")
        self.geometry("800x600")
        self.create_ui(str_to_test)

    def create_ui(self, str_to_test):
        self.main_frame = customtkinter.CTkFrame(self)
        self.main_frame.pack(fill="both", expand=True)
        self.label_with_chinese = customtkinter.CTkLabel(self.main_frame, text=str_to_test)
        self.label_with_chinese.pack(fill="both", expand=True)

    def confg(self):
        self.label_with_chinese.configure(text="你好我是一個有趣的標簽")

str_with_issue = "沒有選擇檔案"
app1 = TestApp(str_with_issue)
app1.mainloop()


str_without_issue = "你好我是一個有趣的標簽"
app2 = TestApp(str_without_issue)
aaa=customtkinter.CTkButton(app2,text="configure",command=app2.confg)
aaa.pack()

app2.mainloop()`

**Comment by felipetesc** [comment]
What is the name of the font you are using? I will try to install on my side. Then, I'll create a test with it.

**Comment by Corporation001** [comment]
> What is the name of the font that you are using? I will try to install on my side. Then, I'll create a test with it.

 If you used the code I edited, it will display"新細明體" in Chinese Windows 11.

**Comment by Corporation001** [comment]
> > What is the name of the font that you are using? I will try to install on my side. Then, I'll create a test with it.
> 
> If you used the code I edited, it will display"新細明體" in Chinese Windows 11.

I have another computer is Windows 11 too. but I modifyed it's default font to "微軟正黑體(Microsoft JhengHei)"
I got the same result.

**Comment by felipetesc** [comment]
Hello. I installed both fonts and unfortunately I didn't encounter any problems again. But this problem is not happening with my pc because I don't use Windows. So I really think someone else with a Windows OS should help you out. This is frustrating, but I don't know what else I can do to help you.

**Comment by Corporation001** [comment]
> def confg(self):
>     self.label_with_chinese.configure(text="你好我是一個有趣的標籤")

Do you mean this code is working fine under Linux? I'm curious how this works under Linux.
But this Bug is actually in Windows...

**Comment by felipetesc** [comment]
> > def confg(self):
> > self.label_with_chinese.configure(text="你好我是一個有趣的標籤")
> 
> Do you mean this code is working fine under Linux? I'm curious how this works under Linux. But this Bug is actually in Windows...

No, at least in the past, it wasn't something possible. Now it's, since CTkLabel implemented the method configure at line 103 inside the file CustomTkinter/customtkinter/widgets/ctk_label.py. Everytime I needed to use configure I called like this: 
```self.label_with_chinese.text_label.configure(text="你好我是一個有趣的標籤")```

**Comment by TomSchimansky** [comment]
For me this: "沒有選擇檔案" works fine with normal tkinter and customtkinter on macOS and Windows 10, so I will move this to discussions...
